tests/test_card.py

```diff
@@ -1,110 +1,158 @@
-
-# going to test the consructor
-from smartcards.card import VXCardObserver, Card4442, CardAT24C, find_card_by_atr, CARD_TYPES, Card
+from .mock_card import MockCard
+from smartcards.card import VXCardObserver, Card4442, CardAT24C, find_card_by_atr, CARD_TYPES, Card, LONG_VALUE_START_CHUNK_INDEX, LONG_VALUE_LENGTH_SIZE, SHORT_VALUE_LENGTH_SIZE
 from smartcard.util import toHexString, toASCIIBytes, toASCIIString
 
-from unittest.mock import patch, MagicMock, Mock
+from unittest.mock import patch, MagicMock, Mock, call
+from typing import Optional
+
+import pytest
+import json
+import os
+import secrets
+import hashlib
+import gzip
+
+CHUNK_SIZE = 250
+
+
+def chunk(*data: bytes) -> bytes:
+    return b''.join(data).ljust(CHUNK_SIZE, b'\0')
+
+
+def build_card_data(short_value: bytes, long_value: Optional[bytes], write_protected=False) -> bytes:
+    card_header = chunk(b'VX.', bytes(
+        [0x02]), write_protected.to_bytes(1, byteorder='big'))
+    card_short_value = chunk(
+        (long_value is not None).to_bytes(1, byteorder='big'),
+        len(short_value).to_bytes(SHORT_VALUE_LENGTH_SIZE, byteorder='big'),
+        hashlib.sha256(short_value).digest(),
+        short_value)
+    encoded_long_value = gzip.compress(long_value or b'')
+    card_long_value = chunk(
+        len(encoded_long_value).to_bytes(
+            LONG_VALUE_LENGTH_SIZE, byteorder='big'),
+        hashlib.sha256(encoded_long_value).digest(),
+        encoded_long_value)
+    return card_header + card_short_value + card_long_value
 
-import pytest, json, os, secrets
 
 # data on card mocked
-CONTENT_BYTES = b'hello1234'
-CARD_BYTES = b'VX.'+bytes([0x01])+bytes([0x00])+bytes([len(CONTENT_BYTES), 0x00, 0x00]) + CONTENT_BYTES
+SHORT_VALUE: bytes = b'hello1234'
+LONG_VALUE: bytes = b'here is a long value I hope you like it'
 
-# copy and modify one byte for write-protected
-write_protected_card_bytearray = bytearray(CARD_BYTES)
-write_protected_card_bytearray[4] = 0x01
-WRITE_PROTECTED_CARD_BYTES = bytes(write_protected_card_bytearray)
+# default card has no long value
+CARD_BYTES = build_card_data(SHORT_VALUE, None)
+EMPTY_CHUNK = chunk()
 
-from .mock_card import MockCard
+# same as default card, but write-protected
+WRITE_PROTECTED_CARD_BYTES = build_card_data(
+    SHORT_VALUE, None, write_protected=True)
+
+# card with short and long values
+CARD_BYTES_WITH_LONG_VALUE = build_card_data(
+    SHORT_VALUE, LONG_VALUE)
+
+
+def flip_byte(data, index):
```

Comment:
should we be testing with a single bit flip rather than a whole byte?

Reply:
Sure, should be easy enough.